packages/swap/src/routes/v2/swap.ts

```diff
@@ -136,12 +151,15 @@ router.get(
       swapId: swapRequest?.nativeId.toString(),
       ...getAssetAndChain(internalSrcAsset, 'src'),
       ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
-      destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+      destAddress:
+        readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress') ??
+        pendingVaultSwap?.destinationAddress,
       srcChainRequiredBlockConfirmations,
       estimatedDurationsSeconds: estimatedDurations?.durations,
       estimatedDurationSeconds: estimatedDurations?.total,
       brokers:
-        beneficiaries?.map(({ account, commissionBps }) => ({ account, commissionBps })) ?? [],
+        beneficiaries?.map(({ account, commissionBps }) => ({ account, commissionBps })) ??
+        [pendingVaultSwap?.brokerFee].concat(pendingVaultSwap?.affiliateFees).filter(Boolean),
```

Comment:
should we normalize the data here to match the schema of the `beneficiaries.map` ?

Reply:
Or we could just transform it at the schema parsing level

Reply:
oh they dont match? 🤔

Reply:
yeah they are slightly different

```
(property) brokers: {
    account: string;
    commissionBps: number;
}[] | ({
    account: string;
    bps: number;
} | undefined)[]
```

Reply:
We could also create another function in the `utils.ts` that will make sure the response is normalized, just like we do for `getCcmParams`, `getFillOrKillParams`, `getDcaParams` etc.

Reply:
oof yes then we should adjust them to match!

Reply:
@GabrielBuragev  nice catch